### ISSUE TITLE
Making the project more object oriented?

### ISSUE THREAD

**Issue by surister** [Pringleman83/SportsBook#14]
The project is looking good bit it's getting a bit big, maybe instead of wrapping everything in functions we can start playing with some classes, that'd make the code insanely more tidy

**Comment by Pringleman83** [comment]
This is something I'm only just learning about so I don't think I could contribute much yet. If you want to try some things out though I'll create a branch for it and learn from what you do.

I was considering having a league class where each selected league would be an instance of it. The class could have special functions then to compare teams, return top scoring teams, lowest scoring teams etc.

I'd be interested to know how you'd plan to approach this and what your ideas are.

**Comment by surister** [comment]
That looks sweet, I'll help you as much as possible.

**Comment by surister** [comment]
> This is something I'm only just learning about so I don't think I could contribute much yet. If you want to try some things out though I'll create a branch for it and learn from what you do.
> 
> I was considering having a league class where each selected league would be an instance of it. The class could have special functions then to compare teams, return top scoring teams, lowest scoring teams etc.
> 
> I'd be interested to know how you'd plan to approach this and what your ideas are.
 
Yeah create a branch called _experiment or something like that and we can settle some foundations

**Comment by surister** [comment]
I think it'd be better if the experiment branch was empty, making the project more object oriented is basically rewriting the codebase in a different fashion (we'll copy paste a lot) but I find it easier having a blank canvas

**Comment by Pringleman83** [comment]
Fully agree. I've deleted most of the files. I've started an 'idea' of how it could work, but I'm only just learning OOC, so it might not be the best approach.
Feel free to add to it or write up some alternative ideas.

**Comment by Pringleman83** [comment]
I've written a couple of classes. One for Leagues and one for Teams. I've complicated things slightly by linking the classes. When a new Team object is instantiated, if necessary a new League object is instantiated too. Checks are performed to see if a League object already exists.

Both classes have class dictionaries keeping links to all League and Team objects. This means all objects are iterable, can be linked to other objects (like a one League to many Teams relationship) and checked for existence.

It's a little complicated (for me at least), but it should help to simplify related code.
